### ISSUE TITLE
Kotlin generation doesn't keep kotlin.collections.List<String>

### ISSUE THREAD

**Issue by rishabh876** [permissions-dispatcher/PermissionsDispatcher#599]
Overview
```
@NeedsPermission(Manifest.permission.CAMERA)
fun aFunction(option: kotlin.collections.List<String>) {
…
}
```

Expected
```
fun Activity.aFunctionWithPermissionCheck(option: kotlin.collections.List<String>) {
…
}
```

I expect both Kotlin List type.

Actual
```
fun Activity.aFunctionWithPermissionCheck(option: java.util.List<String>) {
…
}
```

Environment
Which library version are you using? : 4.3.0

**Comment by hotchemi** [comment]
@rishabh876 Thx. This actually derives from KotlinPoet and we've added some similar workarounds. Would you mind send us PR?

https://github.com/permissions-dispatcher/PermissionsDispatcher/blob/3df06674b4232403bcc548eabca4b12bd9a5ffb8/processor/src/main/kotlin/permissions/dispatcher/processor/util/Extensions.kt#L124

**Comment by cmoijulien** [comment]
@hotchemi 
Can you re-open this issues. I have the same problem with the last release 4.5.0
`Type mismatch: inferred type is java.util.List<String> but kotlin.collections.List<String> was expected`

**Comment by johngoren** [comment]
Same here.

**Comment by hotchemi** [comment]
gonna take a look 🙇

**Comment by hotchemi** [comment]
I think this issue is addressed in 4.6.0.
